File: Java/SingleLinkedLists/Node.java
//	Stack Node
//	Author Name: Jason Disher
//	Student No: c3185333
//	Last Modified: 24/2/2022


public class Node<E>
{
	private E payload;
	private Node<E> next;
	

		//Constructor
	public Node()
	{
		payload=null;
		next=null;
	}
	public Node(E data)
	{
		payload=data;
		next=null;
	}
		//Mutators
	public void setNext(Node<E> node)
	{
		next=node;
	}
	public void setPayload(E data)
	{
		payload=data;
	}
		//Queries
	public E getPayload()
	{
		return payload;
	}
	public Node<E> next()
	{
		return next;
	}
}

File: C++/LinkedList/Nodes/StringNode/node.cpp
// Programmer:  Jason Disher
// Student Number: c3185333
// Assignment 1
// Last modified:  24/8/2017

#include "node.h"

namespace piidish_assignment1
{
			// Typedef
			typedef std::string value_type;
					// Constructor
			node::node(const value_type& init_data, node* init_next, node* init_previous)
				{
					data=init_data;
					next=init_next;
					previous=init_previous;
				}
					//Destructor
			node::~node()
			{
				next=NULL;
				previous=NULL;
			}
					// Mutators
			void node::set_data(const value_type& new_data)
			{
				data=new_data;
			}
			void node::set_next(node* next_link)	
			{
				next=next_link;
			}
			void node::set_previous(node* previous_link)
			{
				previous=previous_link;
			}
					//Constant query functions
			value_type node::get_data() const	
			{
				return data;
			}
			
			const node* node::get_next() const
			{
				return next;
			}
			node* node::get_next()
			{
				return next;
			}
			const node* node::get_previous() const
			{
				return previous;
			}
			node* node::get_previous()
			{
				return previous;
			}

}
